rosplan_sensing_interface/example.yaml

```diff
@@ -0,0 +1,51 @@
+functions:
+    - $(find rosplan_sensing_interface)/example.py
+    #- /home/gerard/Dropbox/rosplan_sensing_interface/example.py
+topics:
+    test:
+        - /topic
+        - std_msgs/String # Not needed, but useful for topics not yet published!
+        #- msg.data == 'TRUE' # value computation
+
+    docked:
+        - params kenny
+        - /mobile_base/sensors/core
+        - kobuki_msgs/SensorState
+        - msg.charger != msg.DISCHARGING
+
+    undocked:
+        - params kenny
+        - /mobile_base/sensors/core
+        - kobuki_msgs/SensorState
+        - msg.charger == msg.DISCHARGING
+
+    robot_at:
+        - params kenny *
+        - /amcl_pose
+        - geometry_msgs/PoseWithCovarianceStamped
+
+    localised:
+        - params kenny
+        - /localised_mock
+        - std_msgs/Bool
+        - msg.data
+
+    somebody_at:
+        - /somebodyat_mock
+        - diagnostic_msgs/KeyValue
+        - '[(msg.key, msg.value == "yes")]'
+
+    is_busy:
+        - params *
+        - /isbusy_mock
+        - diagnostic_msgs/KeyValue
+        - '[(msg.key, msg.value == "yes")]'
+
+services:
+    docked:
+        - params kenny # List of parameters (optional if no parameters!). must be completely instantiated
+        - /test_service # Service
+        - std_srvs/SetBool # Srv type
+        - 10 # Time between calls in seconds
+        - req code # (Optional if in script) Request creation code. I.e.: "req SetBoolRequest(data=True)"
+        - 'res.lala < ee' # (Optional if in script) operation to get the result
```

Comment:
Why not use a map for parameters rather than a list? It would look like this:
docked:
  params:
    - kenny
  service: /test_service
  srv_type: std_srvs/Bool
  time_between_calls: 10

and so on.
The parameters could then be given meaningful keys that will make maintenance easier. Currently it is hard to see what the "10" refers to unless there is an additional comment to remind you.

Reply:
Github has removed my indentation (probably because I didn't put it in a code block) but you get the idea.

Reply:
I like the idea, will try now.